phoenix/app/api_products/lib/api_products_web/controllers/products_controller.ex

```diff
@@ -0,0 +1,56 @@
+defmodule ApiProductsWeb.ProductsController do
+  use ApiProductsWeb, :controller
+
+  alias ApiProducts.Management
+
+  action_fallback ApiProductsWeb.FallbackController
+
+  plug :product_exists? when action in [:show, :update, :delete]
+
+  def index(conn, _params) do
+    products = Management.list_products()
+    render(conn, "index.json", products: products)
+  end
+
+  def create(conn, %{"product" => product_params}) do
+    case Management.create_product(product_params) do
+      {:ok, result} ->
+        conn
+        |> put_status(:created)
+        |> render("show.json", product: result)
+      {:error, result} ->
+        {:error, result}
+    end
+  end
+
+  def show(conn, %{"id" => id}) do
+    render(conn, "show.json", product: conn.assigns[:product])
+  end
+
+  def update(conn, %{"id" => id, "product" => product_params}) do
+    case conn.params["id"] |> Management.get_product() |> Management.update_product(product_params) do
+      {:ok, _} -> json(conn, %{:ok => "The product with id #{conn.params["id"]} has updated!"})
```

Comment:
Aqui (e no delete) retornar um 204 é suficiente, se quiser manter a resposta pode alterar o formato para:

`json(conn, %{message: "The product with id #{conn.params["id"]} has updated!"})`

atualmente está um pouco estranha:
```
{
    "ok": "The product with id 61bb52e87200b501ca035e8f has been updated!"
}
```